### PR TITLE
54 create formsubmissiontable component

### PR DESCRIPTION
**Pull request by shivarn05**
# Description
 Created_ a formSubmissionTable component that takes in a FormSubmissionResponse[] array and outputs a table 

## Relevant issue(s)

#54 Creating the const columns was fairly simple, however, creating the const rows required heavy research and debugging. Also, in the issue, the styling of the table was not specified, such as the height and width of table.

## Questions

Could I have done the const rows with a for loop? I found the .map function to be the easiest and most efficient method.

## Type of change

-  New feature (non-breaking change which adds functionality)

# Checklist:

- My code follows the style guidelines of this project
-  I have performed a self-review of my code
-  I have commented my code, particularly in hard-to-understand areas
-  I have made corresponding changes to the documentation
-  My changes generate no new warnings
-  I have added tests that prove my fix is effective or that my feature works
-  I have assigned reviewers to this PR

### CODE REVIEW

File: apps/form-builder/src/components/FormSubmissions/FormSubmissionTable/index.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      if (typeof submission_element.responderEmail === 'string') {
+        if (!submission_element.createdAt) {
+          responderEmail = 'Anonymous';
+        } else {
+          responderEmail = submission_element.responderEmail;
+        }
+      } else {
+        //if not a string, return null
+        return null;
+      }
```
Comment (LSmith2174) [inline apps/form-builder/src/components/FormSubmissions/FormSubmissionTable/index.tsx:27]: For this section, the logic should be if there's not a valid email then set responderEmail to 'Anonymous'. We can do this with this if statement: `if (!submission_element.responderEmail)`.  This also accounts for empty strings.

File: apps/form-builder/src/components/FormSubmissions/FormSubmissionTable/index.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+          //if successfully parsed, then createdAtDate equals the parsedDate.
+          createdAtDate = new Date(parsedDate);
+        } else {
+          //if parsing fails, return null
+          return null;
+        }
+      } else {
+        //if neither a string or Date object, return null
+        return null;
+      }
```
Comment (LSmith2174) [inline apps/form-builder/src/components/FormSubmissions/FormSubmissionTable/index.tsx:45]: This can just be `createdAtDate = new Date(submission_element.createdAt)`. We are guaranteed to have a date string from the database here, so that is why it doesn't need to be checked. Good job for knowing you had to parse the date! :)

File: apps/form-builder/src/components/FormSubmissions/FormSubmissionTable/index.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      //return the const rows object with submission id and two fields, col1 and col2
+      return {
+        id: submission_element._id,
+        col1: responderEmail,
+        col2: createdAtDate.toLocaleDateString(),
+      };
+    })
+
+    //filter out any null element and pass the rows array to GridValidRowModel[]
+    .filter((row) => row !== null) as GridValidRowModel[];
```
Comment (LSmith2174) [inline apps/form-builder/src/components/FormSubmissions/FormSubmissionTable/index.tsx:56]: This can be removed because we never return null from the map.

File: apps/form-builder/src/components/FormSubmissions/FormSubmissionTable/index.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+  //define the const columns
+  const columns: GridColDef[] = [
+    { field: 'col1', headerName: 'Responder Email', width: 300 },
+    { field: 'col2', headerName: 'Date Submitted', width: 150 },
+  ];
+  //return the table. the height and width of the table can be adjusted easily from here.
+  return (
+    <div style={{ height: 800, width: '100%' }}>
```
Comment (LSmith2174) [inline apps/form-builder/src/components/FormSubmissions/FormSubmissionTable/index.tsx:65]: We can remove this styling so that the one using the component can say how big it is instead.

File: apps/form-builder/src/components/FormSubmissions/FormSubmissionTable/index.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+      <DataGrid rows={rows} columns={columns} />
+    </div>
+  );
+}
+
+//test for the FormSubmissionTable Component
```
Comment (LSmith2174) [inline apps/form-builder/src/components/FormSubmissions/FormSubmissionTable/index.tsx:71]: Remove test code.

**Review by LSmith2174 — CHANGES_REQUESTED** [review]
Sorry you'll have to resolve the merge conflicts again 😅. Great job on this, I can see you thought about edge cases here. Just a few changes to polish off this code! Thanks :) 👍

**Comment by shivarn05** [comment]
Changes have been made, ready to merge now after checked again.

**Review by LSmith2174 — APPROVED** [review]
Looks good! great job 😁👍
